client/lib/interval/index.js

```diff
@@ -0,0 +1,64 @@
+import React, { PropTypes } from 'react';
+
+/**
+ * Calls a given function on a given interval
+ */
+export default React.createClass( {
+	displayName: 'Interval',
+
+	propTypes: {
+		onTick: PropTypes.func.isRequired,
+		milliseconds: PropTypes.number,
+		leading: PropTypes.bool,
+		pauseWhenHidden: PropTypes.bool,
+		children: PropTypes.element
+	},
+
+	getDefaultProps: () => ( {
+		milliseconds: 30000,
+		leading: true,
+		pauseWhenHidden: true
+	} ),
+
+	getInitialState: () => ( {
+		timer: null
+	} ),
+
+	componentDidMount() {
+		this.start()
+	},
+
+	componentWillUnmount() {
+		this.stop()
+	},
+
+	componentDidUpdate() {
+		this.start();
+	},
+
+	run() {
+		clearTimeout( this.state.timer );
+
+		if ( document.hidden && this.props.pauseWhenHidden ) {
+			return this.setState( { timer: null } );
+		}
+
+		this.setState( { timer: setTimeout( this.run, this.props.milliseconds ) } );
+
+		this.props.onTick();
```

Comment:
I've run into problems before with invoking a callback after setting state where the callback depends on the state. The sure fire way for this to work as intended is to do 
```js
this.setState( { timer: setTimeout( this.run, this.props.milliseconds ) }, () => this.props.onTick() );
```